### PR TITLE
SourceInfo: simplify the common case for makeMessage, print it differently

### PR DESCRIPTION
**Pull request by mwachs5**
### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- API modification

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

#### Release Notes

Add a default to SourceInfo.makeMessage() so that users don't always have to write makeMessage(x => x). Change the toString of SourceLine so that it puts a colon instead of a space between the file name and line number, which makes it possible to click-to-source in editors like VSCode.

### Reviewer Checklist (only modified by reviewer)
- [x] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [x] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: core/src/main/scala/chisel3/experimental/SourceInfo.scala [hunk 1]
```diff
@@ -51,9 +51,13 @@ case object DeprecatedSourceInfo extends NoSourceInfo
   * @note A column == 0 indicates no column
   */
 case class SourceLine(filename: String, line: Int, col: Int) extends SourceInfo {
-  def makeMessage(f: String => String): String = f(s"@[${this.serialize}]")
+  def makeMessage(f: String => String = x => x): String = f(s"@[${this.prettyPrint}]")
   def filenameOption: Option[String] = Some(filename)
 
+  def prettyPrint: String = {
```
Comment (mwachs5) [inline core/src/main/scala/chisel3/experimental/SourceInfo.scala:57]:
```suggestion
  private def prettyPrint: String = {
```

**Review by sequencer — APPROVED** [review]
LGTM

**Review by jackkoenig — APPROVED** [review]
LGTM!

You have some failing tests though, they look like tests that just need to be updated for the new format:
```
[info] *** 4 TESTS FAILED ***
[error] Failed tests:
[error] 	chiselTests.experimental.hierarchy.InstantiateSpec
[error] 	chiselTests.WhenSpec
[error] 	chiselTests.simulator.SimulatorSpec
```

**Review by dtzSiFive — APPROVED** [review]
LGTM

**Pull request merged**